include/OsmAndCore/Utilities.h

```diff
@@ -889,6 +889,38 @@ namespace OsmAnd
         static LatLon rhumbDestinationPoint(LatLon latLon, double distance, double bearing);
         static std::pair<int, int> calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
                                                                           int xBase, int yBase, bool ignoreNotEnoughPrecision);
+        template<typename T>
+        inline static T computeDirection(
+            const T& start,
+            const T& end
+        )
+        {
+            auto length = std::sqrt((end.x - start.x)*(end.x - start.x) + (end.y - start.y)*(end.y - start.y));
```

Comment:
`(end - start).norm()` ?

Reply:
1. Why do we need templates at all? computation varies very much between float and int (doesn't take int account overflows and other pitfalls).
And most important we don't need generic method.

2!! There is no need to have it in Utilities - only slows down calculation

Reply:
Template have been omitted.

Reply:
> `(end - start).norm()` - Ok, really.

I as a bit confused when didn't found function like `normalize()` ([Ref](https://glm.g-truc.net/0.9.4/api/a00131.html#ga68b931c228750eeba844c273c54ca43d)) -- @alexey-pelykh , do you have concerns regarding to add this function to the `Point` - or it was unnecessary before?

Reply:
No objections, but `PointT normalized() const` if you intend to compute normalized one and return it or `Point& normalize()` if you plan to alter the object itself.

There was need to compute normalized vectors before since I'm trying to avoid sqrt() and do the computations in square form if needed. The `norm()` itself is used only in 2 places, while `squareNorm()` is used in all the rest.

Reply:
Cool.
`PointT normalized() const` - have been added.